coderd/templateversions.go

```diff
@@ -63,6 +63,40 @@ func (api *API) templateVersion(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, convertTemplateVersion(templateVersion, convertProvisionerJob(job), user))
 }
 
+// @Summary Patch template version by ID
+// @ID patch-template-version-by-id
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Templates
+// @Param templateversion path string true "Template version ID" format(uuid)
+// @Param request body codersdk.PatchTemplateVersionRequest true "Patch template version request"
+// @Success 200 {object} codersdk.TemplateVersion
+// @Router /templateversions/{templateversion} [patch]
+func (api *API) patchTemplateVersion(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	templateVersion := httpmw.TemplateVersionParam(r)
+	var params codersdk.PatchTemplateVersionRequest
+	if !httpapi.Read(ctx, rw, r, &params) {
+		return
+	}
+	if params.Name == "" {
+		params.Name = templateVersion.Name
+	}
+	templateVersion, err := api.Database.UpdateTemplateVersionByID(ctx, database.UpdateTemplateVersionByIDParams{
+		ID:   templateVersion.ID,
+		Name: params.Name,
+	})
```

Comment:
This is potentially unsafe in the future if we keep adding more modifiable parameters. WDYT about this path:
1. Use existing `TemplateVersion` to create `UpdateTemplateVersionByIDParams` from it.
2. Apply `PatchTemplateVersionRequest` properties to `UpdateTemplateVersionByIDParams`.
3. Call `UpdateTemplateVersionByID(params)`.

Reply:
Makes a lot of sense.